### PR TITLE
activesupport: Add types for to_fs

### PR DESCRIPTION
**Pull request by dak2**
- ActiveSupport::NumericWithFormat
  - https://api.rubyonrails.org/v7.0.8.6/classes/ActiveSupport/NumericWithFormat.html#method-i-to_fs
- ActiveSupport::RangeWithFormat
  - https://api.rubyonrails.org/v7.0.8.6/classes/ActiveSupport/RangeWithFormat.html#method-i-to_fs
- Array
  - https://api.rubyonrails.org/v7.0.8.6/classes/Array.html#method-i-to_fs
- Date
  - https://api.rubyonrails.org/v7.0.8.6/classes/Date.html#method-i-to_fs
- DateTime
  - https://api.rubyonrails.org/v7.0.8.6/classes/DateTime.html#method-i-to_fs
- Time
  - https://api.rubyonrails.org/v7.0.8.6/classes/Time.html#method-i-to_fs

### CODE REVIEW

File: gems/activesupport/7.0/activesupport-7.0.rbs [hunk 1]
```diff
@@ -138,3 +138,242 @@ end
...
+  #  1234567890123456789.to_fs(:human)                       # => "1230 Quadrillion"
+  #  489939.to_fs(:human, precision: 2)                      # => "490 Thousand"
+  #  489939.to_fs(:human, precision: 2, round_mode: :down)   # => "480 Thousand"
+  #  489939.to_fs(:human, precision: 4)                      # => "489.9 Thousand"
+  #  1234567.to_fs(:human, precision: 4,
+  #                   significant: false)                             # => "1.2346 Million"
+  #  1234567.to_fs(:human, precision: 1,
+  #                   separator: ',',
+  #                   significant: false)                             # => "1,2 Million"
+  def to_fs: (?Symbol format, *untyped) -> String
```
Comment (tk0miya) [inline gems/activesupport/7.0/activesupport-7.0.rbs:262]: The second argument is a Hash object or kwargs.
https://github.com/rails/rails/blob/v7.0.8.7/activesupport/lib/active_support/core_ext/numeric/conversions.rb#L111

How about this?

```suggestion
  def to_fs: (?Symbol format, **untyped) -> String
```
```suggestion
  def to_fs: (?Symbol format, Hash[untyped, untyped]?) -> String
```

either is okay to me.

File: gems/activesupport/7.0/activesupport-7.0.rbs [hunk 2]
```diff
@@ -138,3 +138,242 @@ end
...
+  # or Proc instance that takes a time argument as the value.
+  #
+  #   # config/initializers/time_formats.rb
+  #   Time::DATE_FORMATS[:month_and_year] = '%B %Y'
+  #   Time::DATE_FORMATS[:short_ordinal]  = ->(time) { time.strftime("%B #{time.day.ordinalize}") }
+  def to_fs: (?Symbol format) -> String
+end
+
+# active_support/core_ext/date_time/conversions.rb
+class Datetime
```
Comment (tk0miya) [inline gems/activesupport/7.0/activesupport-7.0.rbs:352]: typo?

```suggestion
class DateTime
```

**Comment by tk0miya** [comment]
Note: `#to_fs` was introduced (renamed) since 7.0.2.

refs: https://github.com/rails/rails/pull/44354

File: gems/activesupport/7.0/activesupport-7.0.rbs (hunk 2 above)
Comment (dak2) [reply gems/activesupport/7.0/activesupport-7.0.rbs]: Oops, my bad! Thanks for noticing. I fixed it.

File: gems/activesupport/7.0/activesupport-7.0.rbs (hunk 1 above)
Comment (dak2) [reply gems/activesupport/7.0/activesupport-7.0.rbs]: I chose `**untyped` that simply statement. Thanks for pointing that out!

**Review by tk0miya — COMMENTED** [review]
APPROVE

Thank you for your contribution! 🎉

**Pull request merged**